render/render.js

```diff
@@ -349,15 +349,7 @@ module.exports = function($window) {
 		}
 	}
 	function isRecyclable(old, vnodes) {
-		if (old.pool != null && Math.abs(old.pool.length - vnodes.length) <= Math.abs(old.length - vnodes.length)) {
-			var oldChildrenLength = old[0] && old[0].children && old[0].children.length || 0
-			var poolChildrenLength = old.pool[0] && old.pool[0].children && old.pool[0].children.length || 0
-			var vnodesChildrenLength = vnodes[0] && vnodes[0].children && vnodes[0].children.length || 0
-			if (Math.abs(poolChildrenLength - vnodesChildrenLength) <= Math.abs(oldChildrenLength - vnodesChildrenLength)) {
-				return true
-			}
-		}
-		return false
+		return old.pool != null && vnodes.length > 0 && vnodes[0] != null && (vnodes[0].key != null ? true : vnodes.length > old.length)
```

Comment:
:+1: for the ternary, and see https://github.com/lhorie/mithril.js/issues/1653#issuecomment-290993488 for some details... I don't really understand the intent behind the previous behavior, but the new function is simpler and seems to work so far (in the tests I have locally).